File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        left=0
        right=0
        maxprofit=0
        while right<len(prices):
            if prices[left]<prices[right]:
                profit=prices[right]-prices[left]
                maxprofit=max(maxprofit,profit)
            else:
                left=right
            right+=1
        return maxprofit
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int right=0;
        int left=0;
        int maxprofit=0;
        while (right<prices.length){
            if(prices[left]<prices[right]){
                int profit=prices[right]-prices[left];
                maxprofit=Math.max(maxprofit,profit);
            }
            else{
                left=right;
            }
            right++;
        }
        return maxprofit;
    }
}